distributed/protocol/compression.py

```diff
@@ -0,0 +1,113 @@
+"""
+Record known compressors
+
+Includes utilities for determining whether or not to compress
+"""
+from __future__ import print_function, division, absolute_import
+
+import logging
+import random
+
+from toolz import identity, partial
+
+from ..config import config
+from ..utils import ignoring
+
+
+compressions = {None: {'compress': identity,
+                       'decompress': identity}}
+
+default_compression = None
+
+
+logger = logging.getLogger(__file__)
+
+
+with ignoring(ImportError):
+    import zlib
+    compressions['zlib'] = {'compress': zlib.compress,
+                            'decompress': zlib.decompress}
+
+with ignoring(ImportError):
+    import snappy
+    compressions['snappy'] = {'compress': snappy.compress,
+                              'decompress': snappy.decompress}
+    default_compression = 'snappy'
+
+with ignoring(ImportError):
+    import lz4
+    compressions['lz4'] = {'compress': lz4.LZ4_compress,
+                           'decompress': lz4.LZ4_uncompress}
+    default_compression = 'lz4'
+
+with ignoring(ImportError):
+    import blosc
+    compressions['blosc'] = {'compress': partial(blosc.compress, clevel=5,
+                                                 cname='lz4'),
+                             'decompress': blosc.decompress}
+
+
+default = config.get('compression', 'auto')
+if default != 'auto':
```

Comment:
If `config['compression'] == 'auto'` then we use the `default_compression` determined above during the import process.